### PR TITLE
GFOP general improvements

### PR DESCRIPTION
**Pull request by AlejandroMC28**
Modified main count function to:
 - Extract study spectra metadata from the GNPS file.
 - Be compatible with the latest Pandas version.
 - Added function to calculate all levels in one go and return a long format dataframe
 - Black formatted

### CODE REVIEW

File: gfop/get_food_counts.py [hunk 1]
```diff
@@ -31,6 +31,32 @@ def get_sample_types(simple_complex='all'):
     return (gfop_metadata[['filename', *col_sample_types]]
             .set_index('filename'))
 
+def get_sample_metadata(gnps_network, all_groups):
+    '''
+    Args:
```
Comment (bittremieux) [inline gfop/get_food_counts.py:41]: **todo:** Use [NumPy-style docstrings](https://numpydoc.readthedocs.io/en/latest/format.html).

File: gfop/get_food_counts.py [hunk 2]
```diff
@@ -31,6 +31,32 @@ def get_sample_types(simple_complex='all'):
     return (gfop_metadata[['filename', *col_sample_types]]
             .set_index('filename'))
 
+def get_sample_metadata(gnps_network, all_groups):
```
Comment (bittremieux) [inline gfop/get_food_counts.py]: **todo:** Start using type hints.

File: gfop/get_food_counts.py [hunk 3]
```diff
@@ -31,6 +31,32 @@ def get_sample_types(simple_complex='all'):
...
+    '''
+    Args:
+        gnps_network(dataframe): Dataframe generated from classical molecular networking
+                                  with study dataset(s) and reference dataset.
+        groups(list): can contain 'G1', 'G2' to denote study spectrum files.
```
Comment (bittremieux) [inline gfop/get_food_counts.py]: **todo:** Make sure the arguments in the docstring match the actual variable names (`groups` vs `all_groups`).

File: gfop/get_food_counts.py [hunk 4]
```diff
@@ -31,6 +31,32 @@ def get_sample_types(simple_complex='all'):
...
+        
+    Return:
+        Dataframe with all the filenames in the study spectrum files and the group they belong to.
+
+    '''
+
+    df_filtered = gnps_network[~gnps_network['DefaultGroups'].str.contains(',')]
+
```
Comment (bittremieux) [inline gfop/get_food_counts.py]: **todo:** Avoid unnecessary blank lines. A single blank line can be used to split up long chunks of code. Two blank lines should not be used within function definitions.

File: gfop/get_food_counts.py (hunk 1 above)
Comment (bittremieux) [inline gfop/get_food_counts.py:41]: **todo:** Start the docstring with a single sentence description of what the function does.

File: gfop/get_food_counts.py [hunk 5]
```diff
@@ -122,14 +143,45 @@ def get_dataset_food_counts(gnps_network, metadata, filename_col,
...
+        levels (integer): Number of levels to calculate food counts for.
+    Return:
+        A long format dataframe with columns: filename, food_type, level, count, group.
+    """
+    gnps_network_df = pd.read_csv(gnps_network, sep='\t')
+    metadata = get_sample_metadata(gnps_network_df, all_groups)
+    
+    all_data = []
+    for level in range(levels + 1):
+        print(f"Processing level {level}")  # Debug statement
```
Comment (bittremieux) [inline gfop/get_food_counts.py]: **todo:** Remove debug print.

**Review by bittremieux — CHANGES_REQUESTED** [review]
Here are a few small changes.

I don't think the code has been formatted by black yet, because this would basically modify everything. Therefore, let's do this as a separate PR, so that it doesn't get mixed with functional changes.

File: gfop/get_food_counts.py (hunk 1 above)
Comment (AlejandroMC28) [reply gfop/get_food_counts.py:41]: resolved

File: gfop/get_food_counts.py (hunk 2 above)
Comment (AlejandroMC28) [reply gfop/get_food_counts.py]: resolved

File: gfop/get_food_counts.py (hunk 4 above)
Comment (AlejandroMC28) [reply gfop/get_food_counts.py]: resolved

File: gfop/get_food_counts.py (hunk 5 above)
Comment (AlejandroMC28) [reply gfop/get_food_counts.py]: resolved

**Pull request merged**